p2p/server/server.go

```diff
@@ -178,6 +180,10 @@ func New(h Host, proto string, handler StreamHandler, opts ...Opt) *Server {
 		opt(srv)
 	}
 
+	if peerInfo, ok := h.(peerinfo.PeerInfo); ok {
+		srv.peerInfo = peerInfo
+	}
```

Comment:
If `peerinfo.PeerInfo` would be a named field in host this would just become:

```suggestion
	srv.peerInfo = h.peerInfo
```

Reply:
`h` is an interface of type `Host` here. The `Host` interface is defined in `p2p/server` package to avoid extra `p2p/server -> p2p` dependency and make tests simpler.

Reply:
Got rid of Server.peerInfo completely